Sources/Player/PlaybackEngine/Internal/Events/AsyncScheduler.swift

```diff
@@ -0,0 +1,23 @@
+protocol AsyncScheduler {
+  static func schedule(code: @escaping () async -> Void) -> AsyncScheduler
+}
+
+protocol AsyncSchedulerFactory {
+  @discardableResult
+  func create(code: @escaping () async -> Void) -> AsyncScheduler
+}
+
+extension Task: AsyncScheduler {
+  static func schedule(code: @escaping () async -> Void) -> AsyncScheduler {
+	Task<Void, Never> {
+	  await code()
+	}
+  }
+}
+
+class TaskAsyncSchedulerFactory: AsyncSchedulerFactory {
```

Comment:
[suggestion] it can be `final`, no?
```suggestion
final class TaskAsyncSchedulerFactory: AsyncSchedulerFactory {
```